src/http/server.rs

```diff
@@ -498,6 +534,7 @@ impl<'a> EspHttpServer<'a> {
 }
 
 impl<'a> Drop for EspHttpServer<'a> {
+    /// Stops and drops the server.
```

Comment:
I'm not sure that it is a common practice to document trait impls. Perhaps rather than doing this here, maybe in the top-level comment (which is indeed very useful) you can mention that the user has to keep around the http server, because dropping it would stop it and destroy everything.